# Helen Yezerets 

### Data Visualization Data Science Fall 2018

Given the following data points, calculate its covariance matrix's eigenvector that corresponds to the first principal component. Write down the first component of the eigenvector. 

(0, 2)

(2, 6)

(1, 4)

(2, 5)

(-1, -1)

You can calculate them by hand, but numpy also provides functions for calculating the covariance matrix and eigenvector & eigenvalues. 

https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.cov.html 

https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.linalg.eig.html 

In [11]:
import numpy as np

A = np.array([[0, 2],[2, 6],[1, 4],[2, 5],[-1, -1]]).T
A

array([[ 0,  2,  1,  2, -1],
       [ 2,  6,  4,  5, -1]])

https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.cov.html 
Calculate covariance matrix

In [12]:

np.cov(A)
x= np.cov(A)
x

array([[1.7 , 3.55],
       [3.55, 7.7 ]])

https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.linalg.eig.html 
Compute the eigenvalues and right eigenvectors of a square array.

In [7]:
from numpy import linalg as LA
w, v = LA.eig(x)

w : (..., M) array

The eigenvalues, each repeated according to its multiplicity. The eigenvalues are not necessarily ordered. The resulting array will be of complex type, unless the imaginary part is zero in which case it will be cast to a real type. When a is real the resulting eigenvalues will be real (0 imaginary part) or occur in conjugate pairs

In [8]:
w

array([0.05215104, 9.34784896])

The 2nd eigenvalue is larger, so this is the PCA1.

v : (..., M, M) array

The normalized (unit “length”) eigenvectors, such that the column v[:,i] is the eigenvector corresponding to the eigenvalue w[i].

In [13]:
v

array([[-0.90704462, -0.42103452],
       [ 0.42103452, -0.90704462]])

The second column corresponds to the second vector, therefore -0.42103452 would be our the first component of the eigenvector. 